app/controllers/admin/site_steps_controller.rb

```diff
@@ -157,6 +157,11 @@ def update
         else
           @site.form_step = 'template'
 
+          if @site.site_template.nil?
```

Comment:
We already have step-aware model validations, so that's no overkill. I think the issue is in line https://github.com/Vizzuality/forest-atlas-landscape-cms/blob/develop/app/models/site.rb#L39 - should be validating starting from step 'template', not 'style'.